nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/test/java/org/apache/nifi/encrypt/KeyedCipherPropertyEncryptorTest.java

```diff
@@ -0,0 +1,105 @@
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertThrows;
+
+public class KeyedCipherPropertyEncryptorTest {
+    private static final Charset DEFAULT_CHARSET = StandardCharsets.UTF_8;
+
+    private static final String PROPERTY = String.class.getName();
+
+    private static final int ENCRYPTED_BINARY_LENGTH = 48;
+
+    private static final KeyedCipherProvider CIPHER_PROVIDER = new AESKeyedCipherProvider();
+
+    private static final EncryptionMethod ENCRYPTION_METHOD = EncryptionMethod.AES_GCM;
+
+    private static final String KEY_ALGORITHM = "AES";
+
+    private static final byte[] STATIC_KEY = StringUtils.repeat("KEY", 8).getBytes(DEFAULT_CHARSET);
+
+    private static final SecretKey SECRET_KEY = new SecretKeySpec(STATIC_KEY, KEY_ALGORITHM);
+
+    private static final SecretKey INVALID_SECRET_KEY = new SecretKeySpec(KEY_ALGORITHM.getBytes(DEFAULT_CHARSET), KEY_ALGORITHM);
+
+    private KeyedCipherPropertyEncryptor encryptor;
+
+    @Before
+    public void setUp() {
+        encryptor = new KeyedCipherPropertyEncryptor(CIPHER_PROVIDER, ENCRYPTION_METHOD, SECRET_KEY);
+    }
+
+    @Test
+    public void testEncryptDecrypt() {
+        final String encrypted = encryptor.encrypt(PROPERTY);
+        final String decrypted = encryptor.decrypt(encrypted);
+        assertEquals(PROPERTY, decrypted);
+    }
+
+    @Test
+    public void testEncryptHexadecimalEncoded() throws DecoderException {
+        final String encrypted = encryptor.encrypt(PROPERTY);
+        final byte[] decoded = Hex.decodeHex(encrypted);
+        assertEquals(ENCRYPTED_BINARY_LENGTH, decoded.length);
+    }
+
+    @Test
+    public void testDecryptEncryptionException() {
+        final String encodedProperty = Hex.encodeHexString(PROPERTY.getBytes(DEFAULT_CHARSET));
+        assertThrows(EncryptionException.class, () -> encryptor.decrypt(encodedProperty));
+    }
+
+    @Test
+    public void testGetCipherEncryptionException() {
+        encryptor = new KeyedCipherPropertyEncryptor(CIPHER_PROVIDER, ENCRYPTION_METHOD, INVALID_SECRET_KEY);
+        assertThrows(EncryptionException.class, () -> encryptor.encrypt(PROPERTY));
+    }
+
+    @Test
+    public void testEqualsHashCode() {
+        final KeyedCipherPropertyEncryptor equivalentEncryptor = new KeyedCipherPropertyEncryptor(CIPHER_PROVIDER, ENCRYPTION_METHOD, SECRET_KEY);
+        assertTrue(encryptor.equals(equivalentEncryptor));
+        assertEquals(encryptor.hashCode(), equivalentEncryptor.hashCode());
+    }
+
+    @Test
+    public void testEqualsHashCodeDifferentSecretKey() {
+        final SecretKey secretKey = new SecretKeySpec(String.class.getSimpleName().getBytes(StandardCharsets.UTF_8), KEY_ALGORITHM);
+        final KeyedCipherPropertyEncryptor differentEncryptor = new KeyedCipherPropertyEncryptor(CIPHER_PROVIDER, ENCRYPTION_METHOD, secretKey);
+        assertFalse(encryptor.equals(differentEncryptor));
```

Comment:
Optionally, this could be simplified to assertNotEquals.

Reply:
As above, the purpose was to call `equals()` explicitly, but will make this change as well.